### PR TITLE
restructure configuration object

### PR DESCRIPTION
**Pull request by wkerzendorf**
Part of the big TARDIS restructure

### CODE REVIEW

File: tardis/io/config_reader.py [hunk 1]
```diff
@@ -425,28 +293,170 @@ def from_config_dict(cls, config_dict, validate=True, config_dirname=""):
...
+                if "time_0" in model_section["structure"]["density"].keys():
+                    time_0 = model_section["structure"]["density"]["time_0"]
+                    if time_0.value <= 0:
+                        raise ValueError(f"Time Specified is Invalid, {time_0}")
 
-        return cls(validated_config_dict)
+    @staticmethod
+    def validate_montecarlo_section(montecarlo_section):
+        """
+        Parse the montecarlo section dictionary
```
Comment (andrewfullard) [inline tardis/io/config_reader.py:409]: Validate

**Review by Rodot- — APPROVED** [review]

File: tardis/io/config_reader.py (hunk 1 above)
Comment (andrewfullard) [inline tardis/io/config_reader.py:409]:
```suggestion
        Validate the montecarlo section dictionary
```

File: tardis/io/config_reader.py [hunk 2]
```diff
@@ -425,28 +293,170 @@ def from_config_dict(cls, config_dict, validate=True, config_dirname=""):
 
             initial_t_inner = plasma_section["initial_t_inner"]
             initial_t_rad = plasma_section["initial_t_rad"]
-            if not initial_t_inner.value >= -1:
+            if initial_t_inner.value < -1:
                 raise ValueError(
                     f"Initial Temperature of Inner Boundary Black Body is Invalid, {initial_t_inner}"
                 )
-            if not initial_t_rad.value >= -1:
+            if initial_t_rad.value < -1:
```
[the remark this answers is not in the document]
Comment (andrewfullard) [reply tardis/io/config_reader.py:300]:
```suggestion
            if initial_t_rad.value < 0:
```

**Review by andrewfullard — APPROVED** [review]

File: tardis/io/config_reader.py [hunk 3]
```diff
@@ -425,28 +293,170 @@ def from_config_dict(cls, config_dict, validate=True, config_dirname=""):
 
             initial_t_inner = plasma_section["initial_t_inner"]
             initial_t_rad = plasma_section["initial_t_rad"]
-            if not initial_t_inner.value >= -1:
+            if initial_t_inner.value < -1:
                 raise ValueError(
                     f"Initial Temperature of Inner Boundary Black Body is Invalid, {initial_t_inner}"
                 )
-            if not initial_t_rad.value >= -1:
+            if initial_t_rad.value < 0:
```
Comment (andrewfullard) [inline tardis/io/config_reader.py:300]:
```suggestion
            if initial_t_rad.value < -1:
```

**Pull request merged**